apps/api/src/app/rate-limiting/guards/throttler.guard.e2e.ts

```diff
@@ -0,0 +1,309 @@
+import { UserSession } from '@novu/testing';
+import { RateLimitHeaderKeysEnum } from './throttler.guard';
+import { expect } from 'chai';
+import { ApiRateLimitCategoryEnum, ApiRateLimitCostEnum, ApiServiceLevelEnum } from '@novu/shared';
+
+const mockSingleCost = 1;
+const mockBulkCost = 5;
+const mockWindowDuration = 5;
+const mockBurstAllowance = 1;
+const mockMaximumFreeTrigger = 5;
+const mockMaximumFreeGlobal = 3;
+const mockMaximumUnlimitedTrigger = 10;
+const mockMaximumUnlimitedGlobal = 5;
+
+process.env.API_RATE_LIMIT_COST_SINGLE = `${mockSingleCost}`;
+process.env.API_RATE_LIMIT_COST_BULK = `${mockBulkCost}`;
+process.env.API_RATE_LIMIT_ALGORITHM_WINDOW_DURATION = `${mockWindowDuration}`;
+process.env.API_RATE_LIMIT_ALGORITHM_BURST_ALLOWANCE = `${mockBurstAllowance}`;
+process.env.API_RATE_LIMIT_MAXIMUM_FREE_TRIGGER = `${mockMaximumFreeTrigger}`;
+process.env.API_RATE_LIMIT_MAXIMUM_FREE_GLOBAL = `${mockMaximumFreeGlobal}`;
+process.env.API_RATE_LIMIT_MAXIMUM_UNLIMITED_TRIGGER = `${mockMaximumUnlimitedTrigger}`;
+process.env.API_RATE_LIMIT_MAXIMUM_UNLIMITED_GLOBAL = `${mockMaximumUnlimitedGlobal}`;
+
+describe('API Rate Limiting', () => {
+  let session: UserSession;
+  const pathPrefix = '/v1/rate-limiting';
+  let request: (
+    path: string,
+    authHeader?: string
+  ) => Promise<Awaited<ReturnType<typeof UserSession.prototype.testAgent.get>>>;
+
+  beforeEach(async () => {
+    process.env.IS_API_RATE_LIMITING_ENABLED = 'true';
```

Comment:
The [E2E tests are all failing in CI](https://github.com/novuhq/novu/actions/runs/7033263232/job/19138794738?pr=4910), it looks as though this runtime flag is not propagating.

Is it necessary to set `IS_API_RATE_LIMITING_ENABLED` to `true` in the `.env/test` for this to succeed? If yes, how do we test for the disabled flag case?